lib/command.js

```diff
@@ -412,6 +412,27 @@ Expecting one of '${allowedValues.join("', '")}'`);
     return this;
   }
 
+  /**
+   * Add hook to await argument and option values before calling action handlers for this command and its nested subcommands.
+   * Useful for asynchronous custom processing (parseArg) of arguments and option-arguments.
+   */
+
+  awaitHook() {
+    this.hook('preAction', async function awaitHook(thisCommand, actionCommand) {
+      actionCommand.processedArgs = await Promise.all(
```

Comment:
`processedArgs` may include an element which is an array for a variadic argument. I am guessing this is not supported by `Promise.all` so might need more complex processing.